### PR TITLE
Remove need for NODE_VERSIONS

### PR DESCRIPTION
**Pull request by caseydavenport**
## Description

Don't need this any more since `calico-node` can report its own version. 

Also do some .gitignore cleaning. 

## Todos
- [ ] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

```release-note
None required
```

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,24 +1,12 @@
...
-venv
-env
+dist/*
 certs
 .idea
 *.pyc
-*~
 .coverage
-cover
-*.tar
```
Comment (tomdee) [inline .gitignore]: Are you sure about removing .tar?

File: .gitignore [hunk 2]
```diff
@@ -1,24 +1,12 @@
...
 *.created-*
-docker
 **/*.sw[pon]
-birdcl
-.vagrant
-gobgp
-calicoctl/calicoctl
-bin
```
Comment (tomdee) [inline .gitignore]: And won't about bin and release? We still put files in those directories that we don't want checked in right?

Comment (caseydavenport) [reply .gitignore]: I don't think we put anything in either of those directories - looks like the binaries are built in `dist` and the release directory hasn't been used in a long time, AFAICT.

File: .gitignore (hunk 1 above)
Comment (caseydavenport) [reply .gitignore]: Nope, we should keep that.

**Pull request merged**